CONTRIBUTING.md

```diff
@@ -44,19 +70,97 @@ Some distributions carry labwc in their repositories or user repositories.
 
 kindly maintain the packages in their respective distro.
 
-Let's keep them informed of new releases and any changes that relate to packaging.
-If you are maintaining a labwc package for another distro feel free to open an issue so we can add you to this list.
+Let's keep them informed of new releases and any changes that relate to
+packaging.  If you are maintaining a labwc package for another distro feel free
+to open an issue so we can add you to this list.
+
+# Coding Style
+
+labwc is written in Drew Devault's preferred [coding style] which is based
+on the [Linux kernel coding style].
+
+The reasons for specifying a style is not that we enjoy creating rules, but
+because it makes reading/maintaining the code and spotting problems much easier.
+
+If you're new to this style and want to get going quickly, either just immitate
+the style around you, or read the summary below and use [checkpatch.pl] to run
+some formatting checks.
+
+## Style summary
+
+Functions are defined as:
+
+```
+return type
+name(parameters...)
+{
+	body
+}
+```
+
+Opening braces go on the same line, except for function definitions, and are
+mandatory even for one-line statements.
+
+```
+if (cond) {
+	...
+}
+```
+
+Put * with the identifier when defining pointers: `char *foo`
+
+Spaces are placed around binary operators `int x = y * (2 + z);`, but not unary
+`foo(--a, -b);`
+
+```
+	/* This is a single-line comment */
+
+	/*
+	 * This is a multi-line comment which is much much much much much much
+	 * longer.
+	 */
+```
+
+`#include` statements at the top of the file are organized by locality (`<>`
+first, then `""`), then alphabetized.
+
+The line length limit is 80 columns, although it is a bit soft.
+
+Tabs are 8 columns wide. Identation with spaces is not used.
```

Comment:
Woops, I am actually using 4 columns, should likely adopt to 8 as well to not accidentally overflow the ~80 cols.

Reply:
I can't believe it :smiley: